endpoints/openrtb2/auction.go

```diff
@@ -1922,6 +1927,28 @@ func validateDevice(device *openrtb2.Device) error {
 	return nil
 }
 
+func validateOrFillCDep(httpReq *http.Request, req *openrtb_ext.RequestWrapper, account *config.Account) error {
+	if account == nil || !account.Privacy.PrivacySandbox.CookieDeprecation.Enabled {
+		return nil
+	}
+
+	secCookieDeprecation := httpReq.Header.Get(secCookieDeprecation)
+	if secCookieDeprecation == "" {
+		return nil
+	}
+	if len(secCookieDeprecation) > 100 {
+		return &errortypes.Warning{
+			Message:     "request.device.ext.cdep must be less than 100 characters",
+			WarningCode: errortypes.SecCookieDeprecationLenWarningCode,
+		}
+	}
+
+	if deviceExt, err := req.GetDeviceExt(); err == nil {
+		deviceExt.SetCDep(secCookieDeprecation)
+	}
```

Comment:
In the validation phase, please bubble up the error from `req.GetDeviceExt()` as this may the first time it's accessed in code. We tend to use this pattern in deeper parts of code past the validation phase.

Reply:
done, added test case as well